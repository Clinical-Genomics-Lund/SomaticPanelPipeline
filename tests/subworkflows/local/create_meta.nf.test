nextflow_workflow {

    name "Test Workflow CHECK_INPUT"
    script "subworkflows/local/create_meta.nf"
    workflow "CHECK_INPUT"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = file("$projectDir/tests-data/gmslymphomav3-0/23KF00457-lymf_gmslymphomav3-0.csv")
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
