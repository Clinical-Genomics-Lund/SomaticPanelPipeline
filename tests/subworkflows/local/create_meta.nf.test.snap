{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "23KF00457-lymf",
                        {
                            "id": "23HL02304-lymf",
                            "group": "23KF00457-lymf",
                            "diagnosis": "other",
                            "type": "N",
                            "clarity_sample_id": "CMD707A2651",
                            "ffpe": "",
                            "purity": false,
                            "sequencing_run": "230829_A01932_0262_AHJ75FDRX3",
                            "reads": "33966764",
                            "clarity_pool_id": "122-341042",
                            "sub": false
                        },
                        "CMD707A2651_122-341042_S59_R1_001.fastq.gz:md5,0606f1dd3dea9e645ddeec86834bc3f6",
                        "CMD707A2651_122-341042_S59_R2_001.fastq.gz:md5,f3005ab6f8fc5a9af5556af429ef9c4f"
                    ],
                    [
                        "23KF00457-lymf",
                        {
                            "id": "23KF00457-lymf",
                            "group": "23KF00457-lymf",
                            "diagnosis": "other",
                            "type": "T",
                            "clarity_sample_id": "CMD707A3447",
                            "ffpe": "FFPE",
                            "purity": false,
                            "sequencing_run": "231024_A01932_0295_AHK5TYDRX3",
                            "reads": "75970669",
                            "clarity_pool_id": "122-355931",
                            "sub": false
                        },
                        "CMD707A3447_122-355931_S39_R1_001.fastq.gz:md5,8c32fd8cbedd363adb0b4270732ba442",
                        "CMD707A3447_122-355931_S39_R2_001.fastq.gz:md5,4385a70d0523138332e17091695f8197"
                    ]
                ],
                "1": [
                    [
                        "23KF00457-lymf",
                        {
                            "id": "23HL02304-lymf",
                            "group": "23KF00457-lymf",
                            "diagnosis": "other",
                            "type": "N",
                            "clarity_sample_id": "CMD707A2651",
                            "ffpe": "",
                            "purity": false,
                            "sequencing_run": "230829_A01932_0262_AHJ75FDRX3",
                            "reads": "33966764",
                            "clarity_pool_id": "122-341042"
                        }
                    ],
                    [
                        "23KF00457-lymf",
                        {
                            "id": "23KF00457-lymf",
                            "group": "23KF00457-lymf",
                            "diagnosis": "other",
                            "type": "T",
                            "clarity_sample_id": "CMD707A3447",
                            "ffpe": "FFPE",
                            "purity": false,
                            "sequencing_run": "231024_A01932_0295_AHK5TYDRX3",
                            "reads": "75970669",
                            "clarity_pool_id": "122-355931"
                        }
                    ]
                ],
                "fastq": [
                    [
                        "23KF00457-lymf",
                        {
                            "id": "23HL02304-lymf",
                            "group": "23KF00457-lymf",
                            "diagnosis": "other",
                            "type": "N",
                            "clarity_sample_id": "CMD707A2651",
                            "ffpe": "",
                            "purity": false,
                            "sequencing_run": "230829_A01932_0262_AHJ75FDRX3",
                            "reads": "33966764",
                            "clarity_pool_id": "122-341042",
                            "sub": false
                        },
                        "CMD707A2651_122-341042_S59_R1_001.fastq.gz:md5,0606f1dd3dea9e645ddeec86834bc3f6",
                        "CMD707A2651_122-341042_S59_R2_001.fastq.gz:md5,f3005ab6f8fc5a9af5556af429ef9c4f"
                    ],
                    [
                        "23KF00457-lymf",
                        {
                            "id": "23KF00457-lymf",
                            "group": "23KF00457-lymf",
                            "diagnosis": "other",
                            "type": "T",
                            "clarity_sample_id": "CMD707A3447",
                            "ffpe": "FFPE",
                            "purity": false,
                            "sequencing_run": "231024_A01932_0295_AHK5TYDRX3",
                            "reads": "75970669",
                            "clarity_pool_id": "122-355931",
                            "sub": false
                        },
                        "CMD707A3447_122-355931_S39_R1_001.fastq.gz:md5,8c32fd8cbedd363adb0b4270732ba442",
                        "CMD707A3447_122-355931_S39_R2_001.fastq.gz:md5,4385a70d0523138332e17091695f8197"
                    ]
                ],
                "meta": [
                    [
                        "23KF00457-lymf",
                        {
                            "id": "23HL02304-lymf",
                            "group": "23KF00457-lymf",
                            "diagnosis": "other",
                            "type": "N",
                            "clarity_sample_id": "CMD707A2651",
                            "ffpe": "",
                            "purity": false,
                            "sequencing_run": "230829_A01932_0262_AHJ75FDRX3",
                            "reads": "33966764",
                            "clarity_pool_id": "122-341042"
                        }
                    ],
                    [
                        "23KF00457-lymf",
                        {
                            "id": "23KF00457-lymf",
                            "group": "23KF00457-lymf",
                            "diagnosis": "other",
                            "type": "T",
                            "clarity_sample_id": "CMD707A3447",
                            "ffpe": "FFPE",
                            "purity": false,
                            "sequencing_run": "231024_A01932_0295_AHK5TYDRX3",
                            "reads": "75970669",
                            "clarity_pool_id": "122-355931"
                        }
                    ]
                ]
            }
        ],
        "timestamp": "2023-12-12T04:18:31.093631"
    }
}